packages/et-mfr/test/Adaptor.test.js

```diff
@@ -0,0 +1,388 @@
+    );
+  });
+
+  it('should get with a query option', async () => {
+    testServer
+      .intercept({
+        path: '/api/Facility',
+        method: 'GET',
+        query: {
+          id: '101',
+        },
+      })
+      .reply(200, { name: 'Health Post', id: 101, isActive: true });
+
+    const finalState = await get('Facility', {
+      query: {
+        id: '101',
+      },
+    })(state);
+
+    expect(finalState.data).to.eql({
+      name: 'Health Post',
+      id: 101,
+      isActive: true,
+    });
+    expect(finalState.data).to.have.property('name', 'Health Post');
+    expect(finalState.data).to.have.property('id', 101);
+    expect(finalState.data).to.have.property('isActive', true);
+  });
+});
+
+describe('post', () => {
+  const items = [...Array(200)].map((_item, index) => ({
+    id: index + 1,
+    name: `Item ${index + 1}`,
+  }));
+  it('should respect showPerPage when set', async () => {
+    testServer
+      .intercept({
+        path: '/api/Facility/GetFacilities',
+        method: 'POST',
+        headers: {
+          Authorization: 'Basic aGVsbG86dGhlcmU=',
+        },
+      })
+      .reply(200, items.slice(0, 2));
+
+    const results = await post('Facility/GetFacilities', { showPerPage: 2 })(
+      state
+    );
+
+    expect(results.data).to.eql(items.slice(0, 2));
+    expect(results.data).to.have.length(2);
+  });
+  it('makes a post request to the right endpoint with options', async () => {
+    testServer
+      .intercept({
+        path: '/api/Facility/ExportCSV',
+        method: 'POST',
+        headers: {
+          Authorization: 'Basic aGVsbG86dGhlcmU=',
+        },
+      })
+
+      .reply(200, 'name,id\nItem 1,1');
+
+    const finalState = await post(
+      'Facility/ExportCSV',
+      { name: 'Item 1' },
+      { parseAs: 'text' }
+    )(state);
+
+    expect(finalState.data).to.eql('name,id\nItem 1,1');
+    expect(finalState.data).to.be.a('string');
+  });
+
+  it('should paginate if showPerPage and is not set', async () => {
+    //first call
+    testServer
+      .intercept({
+        path: '/api/Facility/GetFacilities',
+        method: 'POST',
+        headers: {
+          Authorization: 'Basic aGVsbG86dGhlcmU=',
+        },
+      })
+      .reply(200, {
+        model: items.slice(0, 100),
+      });
+    //second call
+    testServer
+      .intercept({
+        path: '/api/Facility/GetFacilities',
+        method: 'POST',
+        headers: {
+          Authorization: 'Basic aGVsbG86dGhlcmU=',
+        },
+      })
+      .reply(200, {
+        model: items.slice(100),
+      });
+
+    const finalState = await post(
+      'Facility/GetFacilities',
+      {},
+      {
+        defaultPageSize: 100,
+        defaultLimit: 200,
+      }
+    )(state);
+
+    expect(finalState.data).to.eql(items);
+    expect(finalState.data).to.have.length(200);
+  });
+
+  it('should not paginate when other data is passed with showPerPage or perNumber', async () => {
```

Comment:
typo on `pageNumber`

Why should this not paginate if I pass `name`? Or `sort`, for example? is this a server-side behaviour?